File: March 6, 2026.py
def navigate_trail(map):

    directions = [
        (0, 1, "R"),
        (1, 0, "D"),
        (0, -1, "L"),
        (-1, 0, "U")
    ]
    result = ""
    start = None
    visited = set()

    for i in range(len(map)):
        for j in range(len(map[i])):
            if map[i][j] == "C":
                start = (i, j)

    x, y = start

    while True:

        visited.add((x, y))

        for dx, dy, move in directions:

            nx = x + dx
            ny = y + dy

            if (
                0 <= nx < len(map) and
                0 <= ny < len(map[nx]) and
                (nx, ny) not in visited
            ):

                if map[nx][ny] == "T" or map[nx][ny] == "G":

                    result += move

                    if map[nx][ny] == "G":
                        return result

                    x, y = nx, ny
                    break

File: March 6, 2026.js
function navigateTrail(map) {

  const directions = [
    { x: 0, y: 1, move: "R" },
    { x: 1, y: 0, move: "D" },
    { x: 0, y: -1, move: "L" },
    { x: -1, y: 0, move: "U" }
  ];

  let start = null;
  let visited = new Set();
  let result = "";
  

  for (let i = 0; i < map.length; i++) {
    for (let j = 0; j < map[i].length; j++) {
      if (map[i][j] === "C") {
        start = [i, j];
      }
    }
  }

  let [x, y] = start;

  while (true) {

    visited.add(x + "," + y);

    for (let dir of directions) {

      let nx = x + dir.x;
      let ny = y + dir.y;

      if (
        map[nx] &&
        map[nx][ny] &&
        !visited.has(nx + "," + ny)
      ) {

        if (map[nx][ny] === "T" || map[nx][ny] === "G") {

          result += dir.move;

          if (map[nx][ny] === "G") {
            return result;
          }

          x = nx;
          y = ny;
          break;
        }
      }
    }
  }
}